portal/src/components/v2/ImageInput/ImageInput.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useCallback, useRef } from "react";
+import styles from "./ImageInput.module.css";
+
+import placeholderIcon from "../../../images/image_input_placeholder_icon.svg";
+import { FormattedMessage } from "@oursky/react-messageformat";
+import { SecondaryButton } from "../SecondaryButton/SecondaryButton";
+
+export interface ImageInputProps {
+  sizeLimitKB?: number;
+
+  value: string | null; // Must be a base64 data url of an image if not null
+  onValueChange?: (imageBase64DataURL: string) => void;
+  onError?: (error: ImageInputError) => void;
+}
+
+export enum ImageInputErrorCode {
+  UNKNOWN = "UNKNOWN",
+  FILE_TOO_LARGE = "FILE_TOO_LARGE",
+}
+
+export class ImageInputError extends Error {
+  code: ImageInputErrorCode;
+  internalError?: unknown;
+
+  constructor(code: ImageInputErrorCode, internalError?: unknown) {
+    super(`image input error: ${code}`);
+    this.code = code;
+    this.internalError = internalError;
+  }
+}
+
+export function ImageInput({
```

Comment:
Add a storybook for this component.

Reply:
I only add story for components which exist in the storybook figma, so I do not add for this component.

Reply:
I think every components under `src/components/v2` should have a corresponding story. The storybook then become a gallery of reusable components.

Reply:
I think our conclusion before is to just include stories in the storybook of component in the design file. So we have to ask designer to add these component designs to create a story.

Reply:
Maybe I will move ImageInput and ColorInput to the project-wizard directory. We can move it back later if we need them in another places. But I think I won't move TextInput, and also there won't be a new story for TextInput, because it is just a lower level component of the TextField.

Reply:
> I think every components under src/components/v2 should have a corresponding story.

This is our conclusion.

Reply:
There are still some exceptions, namely `TextInput` (=TextField, with lower level interface), `Toast` (no ui), `ThemeProvider` (no ui).
Will add stories for
- FormField
- ImageInput
- ColorInput